forest/shared/src/cli/snapshot_fetch.rs

```diff
@@ -300,6 +405,32 @@ fn filename_from_url(url: &Url) -> anyhow::Result<String> {
     }
 }
 
+fn get_snapshot_name(network: &str, server: &str, filename: &str) -> anyhow::Result<String> {
+    let pattern = Regex::new(
+        r"(?P<height>\d+)_(?P<date>\d{4}_\d{2}_\d{2})T(?P<time>\d{2}_\d{2}_\d{2})Z.car$",
+    )
+    .unwrap();
+    if let Some(captures) = pattern.captures(filename) {
+        let date = Date::parse(
+            captures.name("date").unwrap().as_str(),
+            &format_description::parse("[year]_[month]_[day]").unwrap(),
+        )
+        .unwrap();
+        let height = captures
+            .name("height")
+            .unwrap()
+            .as_str()
+            .parse::<i64>()
+            .unwrap();
+        Ok(format!(
+            "{server}_snapshot_{network}_{}_height_{height}.car",
+            date.format(&format_description::parse("[year]-[month]-[day]").unwrap())?
+        ))
```

Comment:
All of these unwraps make me nervous. Please add a quickcheck test that fuzzes this function with random filenames and checks that nothing panics.

Reply:
fixed